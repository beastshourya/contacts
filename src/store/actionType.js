export const ADDCONTACTS = "Contacts";
export const DELETES = "Deletes";
export const UPDATE = "Update";
