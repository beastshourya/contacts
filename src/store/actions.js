import { ADDCONTACTS, EDIT, DELETES, UPDATE } from "./actionType";

export const Contacts = (contact) => ({
  type: ADDCONTACTS,
  payload: contact,
});

export const deletes = (key) => ({
  type: DELETES,
  key: "",
});
export const Update = (contact) => ({
  type: UPDATE,
  payload: contact,
});
