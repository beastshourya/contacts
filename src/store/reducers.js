import { ADDCONTACTS, DELETES, SEARCH, UPDATE } from "./actionType";

const initialState = {
  myContacts: [],
  filteredContacts: [],
};

export const mainReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADDCONTACTS:
      return {
        ...state,
        myContacts: [...state.myContacts, action.payload],
        filteredContacts: [...state.myContacts, action.payload],
      };

    case DELETES:
      return {
        ...state,
        myContacts: state.myContacts.filter((item) => item.key !== action.key),
        filteredContacts: state.myContacts.filter(
          (item) => item.key !== action.key
        ),
      };
    case UPDATE:
      return {
        ...state,
        myContacts: state.myContacts.map((contact) =>
          contact.key == action.payload.key ? action.payload : contact
        ),

        filteredContacts: state.myContacts.map((contact) =>
          contact.key == action.payload.key ? action.payload : contact
        ),
      };
    case SEARCH:
      return {
        ...state,
        filteredContacts: state.myContacts.filter((item) => {
          return item.name.toLowerCase().includes(action.payload.toLowerCase());
        }),
        filteredContacts: state.myContacts.filter((item) => {
          return item.lastName
            .toLowerCase()
            .includes(action.payload.toLowerCase());
        }),
      };

    default:
      return state;
  }
};
