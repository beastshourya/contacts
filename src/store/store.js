import { createStore } from "redux";

import { mainReducer } from "./reducers";

import { AsyncStorage } from "react-native";
import { persistStore, persistReducer } from "redux-persist";

const persistConfig = {
  key: "persistedReducer",

  storage: AsyncStorage,
};

const persistedReducer = persistReducer(persistConfig, mainReducer);

const store = createStore(persistedReducer);

let persistor = persistStore(store);

export { store, persistor };
