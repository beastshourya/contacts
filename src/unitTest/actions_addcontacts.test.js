import { mainReducer } from "../store/reducers";
import { initialState } from "../store/reducers";

import { setName } from "../store/actions";

it("setName()", () => {
  expect(mainReducer(initialState, setName())).toMatchSnapshot();
});
