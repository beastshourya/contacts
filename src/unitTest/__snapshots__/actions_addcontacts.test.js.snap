// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setName() 1`] = `
{
  "email": "",
  "key": "",
  "myContacts": [],
  "name": "",
  "phone_no": "",
}
`;
